## T-BAS v2.1 (2019)
*   Multi-locus placement of unknown taxa sequences with user-customizable metadata on eleven included reference trees
*   Multi-locus placement on user-uploaded reference tree, multi-locus sequence alignments and metadata
*   Standardization of phylogenetic placement and metadata output using the extended PhyloXML format
*   Perform cumulative placement of unknown taxa sequences on reference tree
*   Pre-filtering and/or classification of unknown nrITS or 16S rDNA barcode sequences using the UNITE fungal nrITS database or SILVA bacterial database
*   Option to select from three phylogeny-based placement options: EPA, backbone constraint tree, and de novo phylogenetic reconstruction using RAxML
*   Option to select specific clades and perform phylogenetic inference across single or multiple alignments with RAxML or network inference using TCS and NetworkX
*   Color editor for synchronizing color-coded attributes across subtrees and networks
*   Display metadata as colorized outer rings in phylogenies
*   Display network graphs with node attributes
*   Save trees, networks and color legends as SVG, PNG or PDF files

## T-BAS v1 (2017)

*	Single-locus placement of unknown taxa sequences and user-customizable metadata on two included trees: fungi (Ascomycota) and bacteria
*	Single-locus placement on a user-provided reference tree, multi-locus sequence alignments and metadata
*	Option to select from two phylogeny-based placement options: EPA and backbone constraint tree
*	Classification of unknown nrITS barcode sequences using the UNITE fungal nrITS database
*	Metadata displayed as colorized outer rings in phylogenies
*	Save trees and color legends as SVG, PNG or PDF files



